sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/PrivateFieldUsedAsLocalVariable.cs

```diff
@@ -36,340 +36,216 @@ namespace SonarAnalyzer.Rules.CSharp
-
-            var localDeclaration = SyntaxFactory.LocalDeclarationStatement(
-                SyntaxFactory.VariableDeclaration(
-                    SyntaxFactory.PredefinedType(
-                        SyntaxFactory.Token(SyntaxKind.IntKeyword)).WithTrailingTrivia(SyntaxFactory.Space),
-                    SyntaxFactory.SeparatedList(symbolsToRewrite.Select(
-                        s => SyntaxFactory.VariableDeclarator(privateFields[s].UniqueName)))));
-
-            var rewrittenNodesBuilder = new HashSet<SyntaxNode>();
-            var newBody = body.ReplaceSyntax(
-                references.Keys,
-                (original, partiallyRewritten) =>
-                {
-                    rewrittenNodesBuilder.Add(original);
-                    return SyntaxFactory.IdentifierName(privateFields[references[original]].UniqueName);
-                },
-                null,
-                null,
-                null,
-                null);
-
-            var newStatements = newBody.Statements.Insert(0, localDeclaration);
-            newBody = newBody.WithStatements(newStatements);
 
-            rewrittenNodes = rewrittenNodesBuilder;
+                return IsLeftSideOfAssignment(fieldReference.Syntax)
+                    || IsOutArgument(fieldReference.Syntax);
 
-            return newBody;
-        }
+                bool IsOutArgument(SyntaxNode syntaxNode) =>
+                    syntaxNode.Parent is ArgumentSyntax argument &&
+                    argument.RefOrOutKeyword.IsKind(SyntaxKind.OutKeyword);
 
-        private static bool ExcludePrivateFieldsBasedOnRewrittenNodes(
-            IImmutableDictionary<ISymbol, PrivateField> privateFields,
-            IImmutableDictionary<SyntaxNode, ISymbol> references,
-            ISet<SyntaxNode> rewrittenNodes)
-        {
-            var partiallyRewrittenSymbols = references
-                .Where(r => !rewrittenNodes.Contains(r.Key))
-                .Select(r => r.Value)
-                .ToHashSet();
+                bool IsLeftSideOfAssignment(SyntaxNode syntaxNode) =>
+                    syntaxNode.Parent is AssignmentExpressionSyntax assignmentExpression &&
+                    assignmentExpression.Left == syntaxNode;
+            }
 
-            foreach (var partiallyRewrittenSymbol in partiallyRewrittenSymbols)
+            private SyntaxNodeWithSymbol<SyntaxNode, ISymbol> GetTopmostSyntaxWithTheSameSymbol(SyntaxNode potentialReference)
             {
-                privateFields[partiallyRewrittenSymbol].Excluded = true;
-            }
+                var referencedSymbol = semanticModel.GetSymbolInfo(potentialReference).Symbol;
 
-            var allSymbolsToRewrite = references.Values.ToHashSet();
+                if (referencedSymbol != null)
+                {
+                    while (potentialReference.Parent != null &&
+                           referencedSymbol.Equals(semanticModel.GetSymbolInfo(potentialReference.Parent).Symbol))

```

Comment:
Don't we have a better way than to loop through parents + parent symbol? This seems overly expansive.